### PR TITLE
Phase: Boss spawn explosive crystals

### PR DESCRIPTION
**Pull request by monikamnc**
This PR adds the phase of the boss that spawns a exploding crystal near the player.

You can test it while playing the boss after doing an attack it will change automaticaly to exploding crystal function and you will see a crystal near you.

Now we can only spawn 5 crystals.

### CODE REVIEW

File: Gameplay/entities/enemies/BossController.cpp [hunk 1]
```diff
@@ -433,6 +443,30 @@ void Hachiko::Scripting::BossController::SpawnCrystals()
...
+
+	GameObject* current_crystal_to_spawn = _explosive_crystals[_current_index_crystals];
+	
+	if (current_crystal_to_spawn == nullptr)
+	{
+		return;
+	}
+
+	float3 emitter_direction = transform->GetFront().Normalized();
+	float random = float(rand() % 10);
```
Comment (Izenz) [inline Gameplay/entities/enemies/BossController.cpp:461]:
```suggestion
	float random_offset = float(rand() % 10);
```

File: Gameplay/entities/enemies/BossController.cpp [hunk 2]
```diff
@@ -433,6 +443,30 @@ void Hachiko::Scripting::BossController::SpawnCrystals()
...
+	float3 emitter_position = transform->GetGlobalPosition() + emitter_direction * random;
```
Comment (Izenz) [inline Gameplay/entities/enemies/BossController.cpp:462]:
```suggestion
	float3 emitter_position = transform->GetGlobalPosition() + emitter_direction * random_offset;
```

**Review by Izenz — APPROVED** [review]
Hey! 

I tested it and it looks good. 

Sometimes whenever a crystal spawns the player clips with the ground and falls outside of the map. This is probably because the spawned crystal clips with the player and when they are moved to not be on top of each other the player ends up below the NavMesh, but I didn't look into it yet. Later on when we polish the bossfight we can set a minimum crystal-player distance so this doesn't happen.

I leave just one variable name change as a suggestion.

Cheers!

**Review by bernatgb — APPROVED** [review]
It also happened to me that the player falls outside of the map.

Also I think that instead of spawning the crystal from nowhere it would be cool to instantiate the crystal under the map and then make it appear as if it's growing from the ground, and when the whole crystal is visible then make it explode.

**Review by Bernatmago — CHANGES_REQUESTED** [review]
Blocking to prevent early merges since @monikamnc said that It would need some

**Review by baransrc — APPROVED** [review]

**Review by Bernatmago — APPROVED** [review]

**Pull request merged**